core/src/main/java/org/ldaptive/SearchRequest.java

```diff
@@ -29,7 +30,7 @@
   private SearchScope searchScope = SearchScope.SUBTREE;
 
   /** Time search operation will block. */
-  private long timeLimit;
+  private Duration timeLimit = Duration.ZERO;
```

Comment:
The ConnectionConfig properties simulated a null by using -1, those properties now allow null.
(Providers specific handling of those properties make their use somewhat nebulous.)
timeLimit doesn't allow null; setting it to zero keeps the same default value.
(Since timeLimit is a defined part of a search request, it's meaning is the same across providers.)